packages/types/package.json

```diff
@@ -0,0 +1,54 @@
+{
+  "name": "@crawlee/types",
```

Comment:
i dont like the name much, given all it contains are the storage related interfaces, maybe something like `@crawlee/abstract-storage` could be better? or we could keep it as a namespace export and export it as `storage` from inside it. still not perfect as it exports only interfaces/types, not some abstract implementation. another idea I had is to have some shared/abstract base in `@crawlee/storage` (what we have there right now will be removed, the storage-local will stay outside of the repo and will not be part of `@crawlee` namespace), this package would 1. export the interfaces, 2. provide common ground for the implementations. so the question is, **do you see some common ground we could extract to reduce duplications to such package?** we would then reuse it in `@apify/storage-local` as well as `@crawlee/memory-storage`. i had this on my mind initially, as to me it feels weird that the storage implementation as quite complex, i would expect very simple CRUD like interface where you can plug basically any kind of storage (let it be file storage, SQL or NoSQL database, even a redis could work).

i know we also export `AllowedHttpMethods` and `Dictionary`, but both are very simple type aliases, we can afford to duplicate them if we want to (and `AllowedHttpMethods` to me sounds like something that should be part of the storage interface anyway)

Reply:
> do you see some common ground we could extract to reduce duplications to such package?

*If* we want to do that, we could restructure the entire `@crawlee/storage` module to act as the client interface and have options for what storage implementation it uses, which would have to extend an abstract storage class that just does CRUD operations (so then `@crawlee/memory-storage` would simply be the CRUD for memory/file system, `@apify/storage-local` would be strictly SQLite, etc).

That does sound a lot better to me (would mean any user can then create wrappers for, say, prisma, or mongodb, cockroachdb, etc.), but I don't know if that's what you're thinking about.

Reply:
Yes you got it right. The question is what to do with the apify client, ideally it would be also just another "adapter" to this `@crawlee/storage` package, but I guess we don't need that initally and we can just keep using the same interface. I would still keep the interfaces we have now (so defined on type level), so it works with the current apify client, and we could migrate later if we want to (or even have "apify platform storage" for crawlee and keep apify client as separate package).